Build fails with GCC 9

Hello,

it turns out that latest version of cqueues does not build using GCC 9.

Example:
https://kojipkgs.fedoraproject.org//work/tasks/3269/32433269/build.log

```

BUILDSTDERR: In file included from /builddir/build/BUILD/cqueues-20171014/src/dns.c:43:
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/dns.c: In function 'res_new':
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:1059:24: error: lvalue required as unary '&' operand
BUILDSTDERR:  1059 | #define dns_opts(...) (&dns_quietinit((struct dns_options)DNS_OPTS_INIT(__VA_ARGS__)))
BUILDSTDERR:       |                        ^
```

or similarly
```
BUILDSTDERR: In file included from /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c:80:
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c: In function 'dns_p_merge':
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:527:16: error: lvalue required as unary '&' operand
BUILDSTDERR:   527 |  dns_rr_i_init(&dns_quietinit((struct dns_rr_i){ 0, __VA_ARGS__ }), (P))
BUILDSTDERR:       |                ^
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:565:53: note: in expansion of macro 'dns_rr_i_new'
BUILDSTDERR:   565 |  for (struct dns_rr_i DNS_PP_XPASTE(i, __LINE__) = *dns_rr_i_new((P), __VA_ARGS__); dns_rr_grep((rr), 1, &DNS_PP_XPASTE(i, __LINE__), (P), &(int){ 0 }); )
BUILDSTDERR:       |                                                     ^~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:567:29: note: in expansion of macro 'dns_rr_foreach_'
BUILDSTDERR:   567 | #define dns_rr_foreach(...) dns_rr_foreach_(__VA_ARGS__)
BUILDSTDERR:       |                             ^~~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c:1672:4: note: in expansion of macro 'dns_rr_foreach'
BUILDSTDERR:  1672 |    dns_rr_foreach(&rr, A, .section = section) {
BUILDSTDERR:       |    ^~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:527:16: error: lvalue required as unary '&' operand
BUILDSTDERR:   527 |  dns_rr_i_init(&dns_quietinit((struct dns_rr_i){ 0, __VA_ARGS__ }), (P))
BUILDSTDERR:       |                ^
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:565:53: note: in expansion of macro 'dns_rr_i_new'
BUILDSTDERR:   565 |  for (struct dns_rr_i DNS_PP_XPASTE(i, __LINE__) = *dns_rr_i_new((P), __VA_ARGS__); dns_rr_grep((rr), 1, &DNS_PP_XPASTE(i, __LINE__), (P), &(int){ 0 }); )
BUILDSTDERR:       |                                                     ^~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:567:29: note: in expansion of macro 'dns_rr_foreach_'
BUILDSTDERR:   567 | #define dns_rr_foreach(...) dns_rr_foreach_(__VA_ARGS__)
BUILDSTDERR:       |                             ^~~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c:1679:4: note: in expansion of macro 'dns_rr_foreach'
BUILDSTDERR:  1679 |    dns_rr_foreach(&rr, B, .section = section) {
BUILDSTDERR:       |    ^~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:527:16: error: lvalue required as unary '&' operand
BUILDSTDERR:   527 |  dns_rr_i_init(&dns_quietinit((struct dns_rr_i){ 0, __VA_ARGS__ }), (P))
BUILDSTDERR:       |                ^
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:565:53: note: in expansion of macro 'dns_rr_i_new'
BUILDSTDERR:   565 |  for (struct dns_rr_i DNS_PP_XPASTE(i, __LINE__) = *dns_rr_i_new((P), __VA_ARGS__); dns_rr_grep((rr), 1, &DNS_PP_XPASTE(i, __LINE__), (P), &(int){ 0 }); )
BUILDSTDERR:       |                                                     ^~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:567:29: note: in expansion of macro 'dns_rr_foreach_'
BUILDSTDERR:   567 | #define dns_rr_foreach(...) dns_rr_foreach_(__VA_ARGS__)
BUILDSTDERR:       |                             ^~~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c:1682:5: note: in expansion of macro 'dns_rr_foreach'
BUILDSTDERR:  1682 |     dns_rr_foreach(&mr, M, .type = rr.type, .section = DNS_S_ALL) {
BUILDSTDERR:       |     ^~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c: In function 'dns_p_dump':
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:527:16: error: lvalue required as unary '&' operand
BUILDSTDERR:   527 |  dns_rr_i_init(&dns_quietinit((struct dns_rr_i){ 0, __VA_ARGS__ }), (P))
BUILDSTDERR:       |                ^
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c:1895:17: note: in expansion of macro 'dns_rr_i_new'
BUILDSTDERR:  1895 |  dns_p_dump3(P, dns_rr_i_new(P, .section = 0), fp);
BUILDSTDERR:       |                 ^~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c: In function 'dns_m_study':
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:527:16: error: lvalue required as unary '&' operand
BUILDSTDERR:   527 |  dns_rr_i_init(&dns_quietinit((struct dns_rr_i){ 0, __VA_ARGS__ }), (P))
BUILDSTDERR:       |                ^
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:565:53: note: in expansion of macro 'dns_rr_i_new'
BUILDSTDERR:   565 |  for (struct dns_rr_i DNS_PP_XPASTE(i, __LINE__) = *dns_rr_i_new((P), __VA_ARGS__); dns_rr_grep((rr), 1, &DNS_PP_XPASTE(i, __LINE__), (P), &(int){ 0 }); )
BUILDSTDERR:       |                                                     ^~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:567:29: note: in expansion of macro 'dns_rr_foreach_'
BUILDSTDERR:   567 | #define dns_rr_foreach(...) dns_rr_foreach_(__VA_ARGS__)
BUILDSTDERR:       |                             ^~~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c:1942:2: note: in expansion of macro 'dns_rr_foreach'
BUILDSTDERR:  1942 |  dns_rr_foreach(&rr, P, .type = DNS_T_OPT, .section = DNS_S_AR) {
BUILDSTDERR:       |  ^~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c: In function 'dns_rr_exists':
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:527:16: error: lvalue required as unary '&' operand
BUILDSTDERR:   527 |  dns_rr_i_init(&dns_quietinit((struct dns_rr_i){ 0, __VA_ARGS__ }), (P))
BUILDSTDERR:       |                ^
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:565:53: note: in expansion of macro 'dns_rr_i_new'
BUILDSTDERR:   565 |  for (struct dns_rr_i DNS_PP_XPASTE(i, __LINE__) = *dns_rr_i_new((P), __VA_ARGS__); dns_rr_grep((rr), 1, &DNS_PP_XPASTE(i, __LINE__), (P), &(int){ 0 }); )
BUILDSTDERR:       |                                                     ^~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.h:567:29: note: in expansion of macro 'dns_rr_foreach_'
BUILDSTDERR:   567 | #define dns_rr_foreach(...) dns_rr_foreach_(__VA_ARGS__)
BUILDSTDERR:       |                             ^~~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/cqueues-20171014/src/lib/dns.c:2702:2: note: in expansion of macro 'dns_rr_foreach'
BUILDSTDERR:  2702 |  dns_rr_foreach(&rr1, P1, .section = rr0->section, .type = rr0->type) {
BUILDSTDERR:       |  ^~~~~~~~~~~~~~
```

This is going to break the package in Fedora 30 which is moving to GCC 9 right now:
https://fedoraproject.org/wiki/Changes/GCC9

--- a/src/lib/dns.h
+++ b/src/lib/dns.h
@@ -156,12 +156,15 @@ DNS_PUBLIC int *dns_debug_p(void);
 
 #define dns_quietinit(...) \
 	DNS_PRAGMA_PUSH DNS_PRAGMA_QUIET __VA_ARGS__ DNS_PRAGMA_POP
-#elif (__GNUC__ == 4 && __GNUC_MINOR__ >= 6) || __GNUC__ > 4
+#elif (__GNUC__ < 9) && ((__GNUC__ == 4 && __GNUC_MINOR__ >= 6) || __GNUC__ > 4)
 #define DNS_PRAGMA_PUSH _Pragma("GCC diagnostic push")
 #define DNS_PRAGMA_QUIET _Pragma("GCC diagnostic ignored \"-Woverride-init\"")
 #define DNS_PRAGMA_POP _Pragma("GCC diagnostic pop")
 
-/* GCC parses the _Pragma operator less elegantly than clang. */
+/*
+ * GCC parses the _Pragma operator less elegantly than clang.
+ * This only works up to GCC 9
+ */
 #define dns_quietinit(...) \
 	__extension__ ({ DNS_PRAGMA_PUSH DNS_PRAGMA_QUIET __VA_ARGS__; DNS_PRAGMA_POP })
 #else

